### PR TITLE
Implement subcommand mechanism

### PR DESCRIPTION
**Pull request by ikesyo**
See https://github.com/Carthage/Carthage/issues/1017 for the context.

> Cargo in particular has a plugin architecture, which led to the formation of a bunch of useful sub-commands.

> I think carthage would greatly benefit from the addition of either support for similar plugins or the addition of similar sub-commands by default.

### CODE REVIEW

File: Commandant/Command.swift [hunk 1]
```diff
@@ -170,8 +170,42 @@ extension CommandRegistry {
...
+
+	/// Finds and executes a subcommand which exists in your $PATH. The executable
+	/// name must be in the form of `command-verb`.
+	///
+	/// - Returns: The exit status of found subcommand or nil.
+	private func executeSubcommandIfExists(executableName: String, verb: String, arguments: [String]) -> Int32? {
+		let subcommand = "\((executableName as NSString).lastPathComponent)-\(verb)"
+		let paths = NSProcessInfo.processInfo()
+			.environment["PATH"]?
+			.characters.split(":")
```
Comment (mdiep) [inline Commandant/Command.swift]: I want there to be a better way to do this… but I guess there's probably not?

Comment (ikesyo) [inline Commandant/Command.swift]: I will update this to use `which -s` and check the exit status.

**Comment by ikesyo** [comment]
Updated. :sparkles: How do you feel now?

**Comment by mdiep** [comment]
This looks and works great. :+1: Thanks!

**Comment by ikesyo** [comment]
Thanks too! :smile:

**Pull request merged**